api/subjects/serializers.py

```diff
@@ -0,0 +1,116 @@
+from rest_framework import serializers as ser
+from rest_framework import exceptions
+
+from framework.auth.core import Auth
+from framework.exceptions import PermissionsError
+
+from api.base.serializers import (
+    JSONAPISerializer,
+    JSONAPIRelationshipSerializer,
+    LinksField,
+    RelationshipField,
+    BaseAPISerializer,
+)
+from osf.exceptions import NodeStateError, ValidationValueError
+
+
+class UpdateSubjectsMixin(object):
```

Comment:
New mixin added which is used in the `TaxonomizableSerializerMixin`, which is then used on four serializers(NodeSerializer, PreprintSerializer, RegistrationSerializer, and CollectionSubmissionSerializer) as well as the `SubjectsRelationshipSerializer`

Handles updating subjects through the resource directly, or through the resource's subjects `self` relationship link.  All error handling consolidated.